p2p/registry.go

```diff
@@ -76,8 +78,8 @@ type StreamInfo struct {
 	RemotePeer peer.ID
 	RemoteAddr ma.Multiaddr
 
-	Local  io.ReadWriteCloser
-	Remote io.ReadWriteCloser
+	Local  manet.Conn
```

Comment:
why are these different interfaces?

Reply:
One is the local port we're listening on, the other is the stream to the other peer. In case of error, I want to be able to reset the stream so I need access to the reset method on `Stream`.

Reply:
Ah, i see. Carry on